storage/src/state/ledger.rs

```diff
@@ -1578,8 +1578,10 @@ impl<N: Network> TransactionState<N> {
     }
 
     // Returns all the record ciphertexts.
-    fn get_ciphertexts(&self) -> impl Iterator<Item = Result<N::RecordCiphertext>> + '_ {
-        self.commitments.keys().map(move |commitment| self.get_ciphertext(&commitment))
+    fn get_ciphertexts(&self) -> impl Iterator<Item = N::RecordCiphertext> + '_ {
+        self.transitions
+            .values()
+            .flat_map(|(_, _, transition)| transition.ciphertexts().cloned().collect::<Vec<N::RecordCiphertext>>())
```

Comment:
is the `collect` needed here?

Reply:
I think so, the `cloned` part gives me the ownership of the ciphertexts in that transition and the `collect` allocates the iterator over those ciphertexts.